pkg/analysis_server/lib/src/services/correction/fix_internal.dart

```diff
@@ -2330,6 +2333,33 @@ class FixProcessor {
     _addFix(DartFixKind.LINT_ADD_OVERRIDE, []);
   }
 
+  void _addLintFixReplaceWithCollectionLiteral() {
+    final InstanceCreationExpression instanceCreation =
+        node.getAncestor((node) => node is InstanceCreationExpression);
+    final InterfaceType type = instanceCreation.staticType;
+    final generics = type.typeArguments;
+    final buffer = new StringBuffer();
+    var addGenerics = false;
```

Comment:
Fixes should generally not do more than necessary. If we want to have a lint rule that creates a lint when the user explicitly specified `dynamic` for all type arguments, the quick fix for that rule can remove them. (But with type inference it might occasionally be necessary to specify `dynamic` as a type argument, so I wouldn't.) This fix should preserve existing type arguments.